server/src/main/java/org/elasticsearch/common/compress/CompressedXContent.java

```diff
@@ -135,6 +137,20 @@ public CompressedXContent(byte[] data) throws IOException {
         this(new BytesArray(data));
     }
 
+    public static CompressedXContent fromJSON(String json) throws IOException {
```

Comment:
Could we have javadoc for this new method?

Reply:
++ added